### PR TITLE
VB-1104 Connect visit contact to visitor list

### PR DESCRIPTION
**Pull request by AledWEvans**
## What does this pull request do?

Connects the visit contact to the visitors list

## What is the intent behind these changes?

to make sure the data is rationalised

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/entity/Visit.kt [hunk 1]
```diff
@@ -95,6 +95,8 @@ data class Visit(
 
   @PostPersist
   fun createReference() {
-    reference = QuotableEncoder(minLength = 8).encode(id)
+    if (reference.isBlank()) {
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/model/entity/Visit.kt:98]: not really part of review,... but check needed to be in!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/DeleteVisitByReferenceTest.kt [hunk 2]
```diff
@@ -48,7 +48,7 @@ class DeleteVisitByReferenceTest : IntegrationTestBase() {
       .save()
 
     visitContactCreator(visit = visitCC, name = "Jane Doe", phone = "01234 098765")
-    visitVisitorCreator(visit = visitCC, nomisPersonId = 123L)
+    visitVisitorCreator(visit = visitCC, nomisPersonId = 123L, visitContact = true)
```
Comment (psoleckimoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/DeleteVisitByReferenceTest.kt:51]: Are there any tests needed where `visitContact` is `false`?

**Review by psoleckimoj — APPROVED** [review]
LGTM, as a non-Kotlin dev 😂

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/DeleteVisitByReferenceTest.kt (hunk 2 above)
Comment (AledWEvans) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/DeleteVisitByReferenceTest.kt:51]: Defiantly needed, we needed to check the values on assert also for update and create... good spot

**Review by psoleckimoj — APPROVED** [review]
LGTM!

**Pull request merged**